independent-projects/tools/codestarts/src/main/java/io/quarkus/devtools/codestarts/CodestartProcessor.java

```diff
@@ -63,10 +69,12 @@ void addBuiltinData() {
     }
 
     void processCodestartDir(final Path sourceDirectory, final Map<String, Object> finalData) {
+        log.debug(() -> "processing dir: " + sourceDirectory.toString());
```

Comment:
People used have debug(format, args) to avoid unnecessary string concatenation in case debug wasn't enabled.

Reply:
Here you are creating lambdas.

Reply:
@aloubyansky but `debug(format, args)` would still process the args (ex: list map and join)

Reply:
Sure, if debug is actually enabled.